### PR TITLE
Added Model Directory in Documentation

### PR DESCRIPTION
**Pull request by gmtiddia**
PR for issue #42

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -30,9 +30,17 @@
...
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
-# import os
-# import sys
-# sys.path.insert(0, os.path.abspath('.'))
+import os
+import sys
+import json
+from pathlib import Path
+sys.path.insert(0, os.path.abspath('.'))
```
Comment (JoseJVS) [inline doc/conf.py:37]: If you are already using ```Path``` then no need to also use ```os.path```
You could replace ```os.path.abspath('.')``` by ```Path().resolve()```

Comment (gmtiddia) [reply doc/conf.py:37]: Thanks! I've replaced it.

Comment (JoseJVS) [reply doc/conf.py:37]: You did the correct modification but apparently you created another conf.py at the root of the repo...

Comment (gmtiddia) [reply doc/conf.py]: You're right, now I've moved it in the doc directory.

File: src/stdp.h [hunk 2]
```diff
@@ -27,6 +27,49 @@
...
+
+
+Parameters
+++++++++++
+
+========== =======  ======================================================
+ tau_plus  ms       Time constant of STDP window, potentiation
+ tau_minus ms       Time constant of STDP window, depression
+ lambda    real     Step size
+ alpha     real     Asymmetry parameter (scales depressing increments as
```
Comment (JoseJVS) [inline src/stdp.h:54]: What do you mean with depressing increments?

File: src/spike_generator.h [hunk 3]
```diff
@@ -32,6 +32,51 @@
...
+
+============== ======= =======================================
+**Parameters:**
+--------------------------------------------------------------
+ spike_times   list    Times in which spikes are emitted
+ spike_heights list    Weight of the spikes emitted
+============== ======= =======================================
+
+Spike times are given in milliseconds, and must be sorted with the
+earliest spike first. All spike times must be strictly in the future.
```
Comment (JoseJVS) [inline src/spike_generator.h:58]: The last sentence does not make sense; strictly what?

File: doc/conf.py [hunk 4]
```diff
@@ -28,11 +28,19 @@
 
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
-# documentation root, use os.path.abspath to make it absolute, like shown here.
+# documentation root, use os.path.abspath to make it absolute.
```
Comment (JoseJVS) [inline doc/conf.py:31]: Must update to mirror the ```Path``` usage

**Review by JoseJVS — APPROVED** [review]

**Review by jhnnsnk — COMMENTED** [review]
Thank you very much for this comprehensive PR!

Minor change-request: I think it would be good to consistently write "conductance-based" with the "-" similar to "current-based".

Besides, I must admit that I find the organization of the Model Directory in NEST (which is here adopted for NEST GPU) a bit confusing as by default the categories are sorted alphabetically without any hierarchy. Are there plans to change this in the future, @jessica-mitchell ? The purpose of this question is just to find out if a follow-up Issue is required.

**Comment by gmtiddia** [comment]
> Minor change-request: I think it would be good to consistently write "conductance-based" with the "-" similar to "current-based".

Thank you! I've put the hyphen on "conductance based" as requested, and I've updated the header of the Python files of the documentation.

**Review by jhnnsnk — APPROVED** [review]

**Pull request merged**